### PR TITLE
Rework `kci user` with new framework

### PR DESCRIPTION
**Pull request by gctucker**
Implement parts of `kci user` with the new framework based on Click and remove the corresponding old implementation code.

### CODE REVIEW

**Comment by JenySadadia** [comment]
@gctucker Is this ready for review?
It would be good to merge these changes before I update user-related API bindings with `fastapi-users` integration.

**Comment by gctucker** [comment]
> @gctucker Is this ready for review?

It is now I've rebased it ;)

File: kci [hunk 1]
```diff
@@ -13,28 +13,12 @@ features.  See the online documentation for more details:
...
-import json
 
-import click
-
-from kernelci.cli import Args, kci
+from kernelci.cli import kci
 from kernelci.cli import (  # pylint: disable=unused-import
     docker as kci_docker,
+    user as kci_user,
 )
```
Comment (JenySadadia) [inline kci:21]: These are unused imports.

Comment (JenySadadia) [reply kci:21]: Seems like it is required to register sub-commands. A comment about it would have been helpful.

**Review by JenySadadia — APPROVED** [review]
Tested OK with early-access instance.

File: kernelci/cli/user.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    configs = kernelci.config.load(config)
+    api_config = configs['api'][api]
+    api = kernelci.api.get_api(api_config)
+    user_token = api.create_token(username, password, scope)
+    click.echo(json.dumps(user_token, indent=indent))
+
+
+@kci_user.group(name='password')
+def user_password():
+    """Manage user passwords"""
```
Comment (JenySadadia) [inline kernelci/cli/user.py:72]: After I added sub-commands to this group, it became `kci user password reset-password`.
I expected it to be `kci password reset-password`.
What are the benefits of adding groups?

Comment (gctucker) [reply kernelci/cli/user.py:72]: Well this was as per the proposed syntax on the GitHub discussion:
https://github.com/kernelci/kernelci-project/discussions/263#discussioncomment-7163554

It means all the user-related commands are grouped under `kci user`.  There might be other passwords for other things, like `kci storage password` or anything else.

**Pull request merged**